taichi/ir/scratch_pad.cpp

```diff
@@ -1,5 +1,22 @@
-#include "scratch_pad.h"
+#include "taichi/ir/scratch_pad.h"
+
+#include "taichi/ir/statements.h"
 
 TLANG_NAMESPACE_BEGIN
 
+std::string ScratchPad::global_to_linearized_local(
```

Comment:
This part is not used for now. It belongs to the old source2source CUDA backend. I left it here since
 - It cleanly explains coordinate conversion between BLS buffer and the block.
 - It may be useful for future source2source backends.